migration-to-python-3-only.md

```diff
@@ -0,0 +1,207 @@
+    high priority as we want pip version to be release and propagate after
+    that.
+  - ASAP: Fix Setuptools to be able to express `requires-python` in `setup.py`.
+  - June 2016 : Release IPython 5.0 LTS
+    - Traitlets "LTS"
+    - ipykernel LTS
+    - ipython_genutils LTS
+  - January 2017 : Release IPython 6.0 Python 3 only. 
+  - July 2017 : Release Notebook 6.0 Python 3 only.
+  - December 2017: IPython 7.0 
+  - December 2017: End of active support IPython 5.x LTS
+  - July 2019 : Complete end of support for IPython 5.x
+
+
+
+# Exposition of the problem
+
+We recognize that many user of IPython and Jupyter still need to use a Python 2
+environment, and will still need to do so for quite some time. Though as an open
+source project, the overhead in maintaining both Python 2 and 3 compatibility at
+the same time is significant, and slow the Jupyter and IPython open-source teams
+to achieve many of our goals.
+
+We also believe that the maturity of Python 3 is sufficient enough, and a
+sufficient number of project in the scientific community are ready to make the
+commitment to drop Python 2 support.
+
+We want to provide a seamless experience to current Python 3 users without
+breaking any workflow and pipeline for Python 2 users. Thus we want in a
+relatively short period of time to remove the burden of support ting Python 2
+support from the core team, while preserving functionality for users. 
+
+We also recognize that companies and laboratories may rely on Python 2, and
+want to offer solution for these too, whether they directly use, or sell
+product relying on Jupyter and IPython.
+
+# IPython 5.x LTS
+
+This part concern the reference kernel implementation of Jupyter, also know as
+IPython. This specific kernel implementation also rely on `ipykernel`, `traitlets`
+and `ipython_genutils`.
+
+We propose that the next major version cycle (5.x) of IPython be the last
+version to support Python 2. As we recognize that Python 2 is supported by the
+CPython core team until 2020, we propose to make the 5.x major cycle a Long
+Term Support version under the following intent:
+
+  - For the 18 month following the release of IPython 5.0, the core team will
+    engage in critical & security bug fix for the 5.x branch, as well as
+    regular release (both of minor and patch revision) of IPython 5.x branch.
+    This apply also to the documentation of the 5.x branch. THe core team will
+    also accept to provide reasonable support requests for the IPython 5.x
+    branch.
+
+
+  - For the 36 month following the release of IPython 5.x, the core team will
+    continue to review critical & security releated patches, as well as
+    improvment to the documentation, and make related release if necessary. The
+    core team will make sure that the 5.x Version of IPython stay visible
+    various web pages mentionning how to get IPython. The core team reserve the
+    right to ignore and or close any support request, request for patch
```

Comment:
This statement feels a bit too formal to me. We're writing a statement of intent, not a legally binding contract.